pkg/render/applicationlayer.go

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/ptr"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/render/common/statik"
+)
+
+const (
+	CalicoSystemNamespace       = "calico-system"
+	EnvoyConfigConfigMapName    = "envoy-config"
+	L7LogCollectorDeamonsetName = "l7-log-collector"
+	L7CollectorContainerName    = "l7-collector"
+	ProxyContainerName          = "envoy-proxy"
+	EnvoyLogsKey                = "envoy-logs"
+	EnvoyConfigKey              = "envoy-config"
+	EnvoyConfigMapKey           = "envoy-config.yaml"
+)
+
+func ApplicationLayer(pullSecrets []*corev1.Secret, installation *operatorv1.InstallationSpec, osType rmeta.OSType,
+	applicationLayer *operatorv1.ApplicationLayer, l7Spec *L7LogCollectionSpec) Component {
+
+	return &ApplicationLayerComponent{
+		pullSecrets:      pullSecrets,
+		installation:     installation,
+		osType:           osType,
+		applicationLayer: applicationLayer,
+		l7Spec:           l7Spec,
+	}
+}
+
+type L7LogCollectionSpec struct {
```

Comment:
I'm not sure there's much benefit in having a struct for just these fields. I think these fields should either just be parameters to the ApplicationLayer function or create a general config struct that the ApplicationLayer function accepts (there's a few examples of this in the other component constructors).

Reply:
removed the struct, passing it as parameters to the ApplicationLayer function